// file-security-sdk/lib/protos/scan_pb.js
"use strict";
// source: protos/scan.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck
Object.defineProperty(exports, "__esModule", { value: true });
var jspb = require('google-protobuf');
var goog = jspb;
var global = (function () {
    if (this) {
        return this;
    }
    if (typeof window !== 'undefined') {
        return window;
    }
    if (typeof global !== 'undefined') {
        return global;
    }
    if (typeof self !== 'undefined') {
        return self;
    }
    return Function('return this')();
}.call(null));
goog.exportSymbol('proto.amaas.scan.v1.C2S', null, global);
goog.exportSymbol('proto.amaas.scan.v1.Command', null, global);
goog.exportSymbol('proto.amaas.scan.v1.S2C', null, global);
goog.exportSymbol('proto.amaas.scan.v1.Stage', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.amaas.scan.v1.C2S = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, proto.amaas.scan.v1.C2S.repeatedFields_, null);
};
goog.inherits(proto.amaas.scan.v1.C2S, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.amaas.scan.v1.C2S.displayName = 'proto.amaas.scan.v1.C2S';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.amaas.scan.v1.S2C = function (opt_data) {
    jspb.Message.initialize(this, opt_data, 0, -1, proto.amaas.scan.v1.S2C.repeatedFields_, null);
};
goog.inherits(proto.amaas.scan.v1.S2C, jspb.Message);
if (goog.DEBUG && !COMPILED) {
    /**
     * @public
     * @override
     */
    proto.amaas.scan.v1.S2C.displayName = 'proto.amaas.scan.v1.S2C';
}
/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.amaas.scan.v1.C2S.repeatedFields_ = [9];
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.amaas.scan.v1.C2S.prototype.toObject = function (opt_includeInstance) {
        return proto.amaas.scan.v1.C2S.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.amaas.scan.v1.C2S} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.amaas.scan.v1.C2S.toObject = function (includeInstance, msg) {
        var f, obj = {
            stage: jspb.Message.getFieldWithDefault(msg, 1, 0),
            fileName: jspb.Message.getFieldWithDefault(msg, 2, ""),
            rsSize: jspb.Message.getFieldWithDefault(msg, 3, 0),
            offset: jspb.Message.getFieldWithDefault(msg, 4, 0),
            chunk: msg.getChunk_asB64(),
            trendx: jspb.Message.getBooleanFieldWithDefault(msg, 6, false),
            fileSha1: jspb.Message.getFieldWithDefault(msg, 7, ""),
            fileSha256: jspb.Message.getFieldWithDefault(msg, 8, ""),
            tagsList: (f = jspb.Message.getRepeatedField(msg, 9)) == null ? undefined : f,
            bulk: jspb.Message.getBooleanFieldWithDefault(msg, 10, false),
            spnFeedback: jspb.Message.getBooleanFieldWithDefault(msg, 11, false),
            verbose: jspb.Message.getBooleanFieldWithDefault(msg, 12, false)
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.amaas.scan.v1.C2S}
 */
proto.amaas.scan.v1.C2S.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.amaas.scan.v1.C2S;
    return proto.amaas.scan.v1.C2S.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.amaas.scan.v1.C2S} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.amaas.scan.v1.C2S}
 */
proto.amaas.scan.v1.C2S.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {!proto.amaas.scan.v1.Stage} */ (reader.readEnum());
                msg.setStage(value);
                break;
            case 2:
                var value = /** @type {string} */ (reader.readString());
                msg.setFileName(value);
                break;
            case 3:
                var value = /** @type {number} */ (reader.readUint64());
                msg.setRsSize(value);
                break;
            case 4:
                var value = /** @type {number} */ (reader.readInt32());
                msg.setOffset(value);
                break;
            case 5:
                var value = /** @type {!Uint8Array} */ (reader.readBytes());
                msg.setChunk(value);
                break;
            case 6:
                var value = /** @type {boolean} */ (reader.readBool());
                msg.setTrendx(value);
                break;
            case 7:
                var value = /** @type {string} */ (reader.readString());
                msg.setFileSha1(value);
                break;
            case 8:
                var value = /** @type {string} */ (reader.readString());
                msg.setFileSha256(value);
                break;
            case 9:
                var value = /** @type {string} */ (reader.readString());
                msg.addTags(value);
                break;
            case 10:
                var value = /** @type {boolean} */ (reader.readBool());
                msg.setBulk(value);
                break;
            case 11:
                var value = /** @type {boolean} */ (reader.readBool());
                msg.setSpnFeedback(value);
                break;
            case 12:
                var value = /** @type {boolean} */ (reader.readBool());
                msg.setVerbose(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.amaas.scan.v1.C2S.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.amaas.scan.v1.C2S.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.amaas.scan.v1.C2S} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.amaas.scan.v1.C2S.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getStage();
    if (f !== 0.0) {
        writer.writeEnum(1, f);
    }
    f = message.getFileName();
    if (f.length > 0) {
        writer.writeString(2, f);
    }
    f = message.getRsSize();
    if (f !== 0) {
        writer.writeUint64(3, f);
    }
    f = message.getOffset();
    if (f !== 0) {
        writer.writeInt32(4, f);
    }
    f = message.getChunk_asU8();
    if (f.length > 0) {
        writer.writeBytes(5, f);
    }
    f = message.getTrendx();
    if (f) {
        writer.writeBool(6, f);
    }
    f = message.getFileSha1();
    if (f.length > 0) {
        writer.writeString(7, f);
    }
    f = message.getFileSha256();
    if (f.length > 0) {
        writer.writeString(8, f);
    }
    f = message.getTagsList();
    if (f.length > 0) {
        writer.writeRepeatedString(9, f);
    }
    f = message.getBulk();
    if (f) {
        writer.writeBool(10, f);
    }
    f = message.getSpnFeedback();
    if (f) {
        writer.writeBool(11, f);
    }
    f = message.getVerbose();
    if (f) {
        writer.writeBool(12, f);
    }
};
/**
 * optional Stage stage = 1;
 * @return {!proto.amaas.scan.v1.Stage}
 */
proto.amaas.scan.v1.C2S.prototype.getStage = function () {
    return /** @type {!proto.amaas.scan.v1.Stage} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};
/**
 * @param {!proto.amaas.scan.v1.Stage} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setStage = function (value) {
    return jspb.Message.setProto3EnumField(this, 1, value);
};
/**
 * optional string file_name = 2;
 * @return {string}
 */
proto.amaas.scan.v1.C2S.prototype.getFileName = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};
/**
 * @param {string} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setFileName = function (value) {
    return jspb.Message.setProto3StringField(this, 2, value);
};
/**
 * optional uint64 rs_size = 3;
 * @return {number}
 */
proto.amaas.scan.v1.C2S.prototype.getRsSize = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};
/**
 * @param {number} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setRsSize = function (value) {
    return jspb.Message.setProto3IntField(this, 3, value);
};
/**
 * optional int32 offset = 4;
 * @return {number}
 */
proto.amaas.scan.v1.C2S.prototype.getOffset = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};
/**
 * @param {number} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setOffset = function (value) {
    return jspb.Message.setProto3IntField(this, 4, value);
};
/**
 * optional bytes chunk = 5;
 * @return {!(string|Uint8Array)}
 */
proto.amaas.scan.v1.C2S.prototype.getChunk = function () {
    return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};
/**
 * optional bytes chunk = 5;
 * This is a type-conversion wrapper around `getChunk()`
 * @return {string}
 */
proto.amaas.scan.v1.C2S.prototype.getChunk_asB64 = function () {
    return /** @type {string} */ (jspb.Message.bytesAsB64(this.getChunk()));
};
/**
 * optional bytes chunk = 5;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getChunk()`
 * @return {!Uint8Array}
 */
proto.amaas.scan.v1.C2S.prototype.getChunk_asU8 = function () {
    return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(this.getChunk()));
};
/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setChunk = function (value) {
    return jspb.Message.setProto3BytesField(this, 5, value);
};
/**
 * optional bool trendx = 6;
 * @return {boolean}
 */
proto.amaas.scan.v1.C2S.prototype.getTrendx = function () {
    return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 6, false));
};
/**
 * @param {boolean} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setTrendx = function (value) {
    return jspb.Message.setProto3BooleanField(this, 6, value);
};
/**
 * optional string file_sha1 = 7;
 * @return {string}
 */
proto.amaas.scan.v1.C2S.prototype.getFileSha1 = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};
/**
 * @param {string} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setFileSha1 = function (value) {
    return jspb.Message.setProto3StringField(this, 7, value);
};
/**
 * optional string file_sha256 = 8;
 * @return {string}
 */
proto.amaas.scan.v1.C2S.prototype.getFileSha256 = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};
/**
 * @param {string} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setFileSha256 = function (value) {
    return jspb.Message.setProto3StringField(this, 8, value);
};
/**
 * repeated string tags = 9;
 * @return {!Array<string>}
 */
proto.amaas.scan.v1.C2S.prototype.getTagsList = function () {
    return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 9));
};
/**
 * @param {!Array<string>} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setTagsList = function (value) {
    return jspb.Message.setField(this, 9, value || []);
};
/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.addTags = function (value, opt_index) {
    return jspb.Message.addToRepeatedField(this, 9, value, opt_index);
};
/**
 * Clears the list making it empty but non-null.
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.clearTagsList = function () {
    return this.setTagsList([]);
};
/**
 * optional bool bulk = 10;
 * @return {boolean}
 */
proto.amaas.scan.v1.C2S.prototype.getBulk = function () {
    return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 10, false));
};
/**
 * @param {boolean} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setBulk = function (value) {
    return jspb.Message.setProto3BooleanField(this, 10, value);
};
/**
 * optional bool spn_feedback = 11;
 * @return {boolean}
 */
proto.amaas.scan.v1.C2S.prototype.getSpnFeedback = function () {
    return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 11, false));
};
/**
 * @param {boolean} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setSpnFeedback = function (value) {
    return jspb.Message.setProto3BooleanField(this, 11, value);
};
/**
 * optional bool verbose = 12;
 * @return {boolean}
 */
proto.amaas.scan.v1.C2S.prototype.getVerbose = function () {
    return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 12, false));
};
/**
 * @param {boolean} value
 * @return {!proto.amaas.scan.v1.C2S} returns this
 */
proto.amaas.scan.v1.C2S.prototype.setVerbose = function (value) {
    return jspb.Message.setProto3BooleanField(this, 12, value);
};
/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.amaas.scan.v1.S2C.repeatedFields_ = [6, 7];
if (jspb.Message.GENERATE_TO_OBJECT) {
    /**
     * Creates an object representation of this proto.
     * Field names that are reserved in JavaScript and will be renamed to pb_name.
     * Optional fields that are not set will be set to undefined.
     * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
     * For the list of reserved names please see:
     *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
     * @param {boolean=} opt_includeInstance Deprecated. whether to include the
     *     JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @return {!Object}
     */
    proto.amaas.scan.v1.S2C.prototype.toObject = function (opt_includeInstance) {
        return proto.amaas.scan.v1.S2C.toObject(opt_includeInstance, this);
    };
    /**
     * Static version of the {@see toObject} method.
     * @param {boolean|undefined} includeInstance Deprecated. Whether to include
     *     the JSPB instance for transitional soy proto support:
     *     http://goto/soy-param-migration
     * @param {!proto.amaas.scan.v1.S2C} msg The msg instance to transform.
     * @return {!Object}
     * @suppress {unusedLocalVariables} f is only used for nested messages
     */
    proto.amaas.scan.v1.S2C.toObject = function (includeInstance, msg) {
        var f, obj = {
            stage: jspb.Message.getFieldWithDefault(msg, 1, 0),
            cmd: jspb.Message.getFieldWithDefault(msg, 2, 0),
            offset: jspb.Message.getFieldWithDefault(msg, 3, 0),
            length: jspb.Message.getFieldWithDefault(msg, 4, 0),
            result: jspb.Message.getFieldWithDefault(msg, 5, ""),
            bulkOffsetList: (f = jspb.Message.getRepeatedField(msg, 6)) == null ? undefined : f,
            bulkLengthList: (f = jspb.Message.getRepeatedField(msg, 7)) == null ? undefined : f,
            sessionId: jspb.Message.getFieldWithDefault(msg, 8, "")
        };
        if (includeInstance) {
            obj.$jspbMessageInstance = msg;
        }
        return obj;
    };
}
/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.amaas.scan.v1.S2C}
 */
proto.amaas.scan.v1.S2C.deserializeBinary = function (bytes) {
    var reader = new jspb.BinaryReader(bytes);
    var msg = new proto.amaas.scan.v1.S2C;
    return proto.amaas.scan.v1.S2C.deserializeBinaryFromReader(msg, reader);
};
/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.amaas.scan.v1.S2C} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.amaas.scan.v1.S2C}
 */
proto.amaas.scan.v1.S2C.deserializeBinaryFromReader = function (msg, reader) {
    while (reader.nextField()) {
        if (reader.isEndGroup()) {
            break;
        }
        var field = reader.getFieldNumber();
        switch (field) {
            case 1:
                var value = /** @type {!proto.amaas.scan.v1.Stage} */ (reader.readEnum());
                msg.setStage(value);
                break;
            case 2:
                var value = /** @type {!proto.amaas.scan.v1.Command} */ (reader.readEnum());
                msg.setCmd(value);
                break;
            case 3:
                var value = /** @type {number} */ (reader.readInt32());
                msg.setOffset(value);
                break;
            case 4:
                var value = /** @type {number} */ (reader.readInt32());
                msg.setLength(value);
                break;
            case 5:
                var value = /** @type {string} */ (reader.readString());
                msg.setResult(value);
                break;
            case 6:
                var values = /** @type {!Array<number>} */ (reader.isDelimited() ? reader.readPackedInt32() : [reader.readInt32()]);
                for (var i = 0; i < values.length; i++) {
                    msg.addBulkOffset(values[i]);
                }
                break;
            case 7:
                var values = /** @type {!Array<number>} */ (reader.isDelimited() ? reader.readPackedInt32() : [reader.readInt32()]);
                for (var i = 0; i < values.length; i++) {
                    msg.addBulkLength(values[i]);
                }
                break;
            case 8:
                var value = /** @type {string} */ (reader.readString());
                msg.setSessionId(value);
                break;
            default:
                reader.skipField();
                break;
        }
    }
    return msg;
};
/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.amaas.scan.v1.S2C.prototype.serializeBinary = function () {
    var writer = new jspb.BinaryWriter();
    proto.amaas.scan.v1.S2C.serializeBinaryToWriter(this, writer);
    return writer.getResultBuffer();
};
/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.amaas.scan.v1.S2C} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.amaas.scan.v1.S2C.serializeBinaryToWriter = function (message, writer) {
    var f = undefined;
    f = message.getStage();
    if (f !== 0.0) {
        writer.writeEnum(1, f);
    }
    f = message.getCmd();
    if (f !== 0.0) {
        writer.writeEnum(2, f);
    }
    f = message.getOffset();
    if (f !== 0) {
        writer.writeInt32(3, f);
    }
    f = message.getLength();
    if (f !== 0) {
        writer.writeInt32(4, f);
    }
    f = message.getResult();
    if (f.length > 0) {
        writer.writeString(5, f);
    }
    f = message.getBulkOffsetList();
    if (f.length > 0) {
        writer.writePackedInt32(6, f);
    }
    f = message.getBulkLengthList();
    if (f.length > 0) {
        writer.writePackedInt32(7, f);
    }
    f = message.getSessionId();
    if (f.length > 0) {
        writer.writeString(8, f);
    }
};
/**
 * optional Stage stage = 1;
 * @return {!proto.amaas.scan.v1.Stage}
 */
proto.amaas.scan.v1.S2C.prototype.getStage = function () {
    return /** @type {!proto.amaas.scan.v1.Stage} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};
/**
 * @param {!proto.amaas.scan.v1.Stage} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setStage = function (value) {
    return jspb.Message.setProto3EnumField(this, 1, value);
};
/**
 * optional Command cmd = 2;
 * @return {!proto.amaas.scan.v1.Command}
 */
proto.amaas.scan.v1.S2C.prototype.getCmd = function () {
    return /** @type {!proto.amaas.scan.v1.Command} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};
/**
 * @param {!proto.amaas.scan.v1.Command} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setCmd = function (value) {
    return jspb.Message.setProto3EnumField(this, 2, value);
};
/**
 * optional int32 offset = 3;
 * @return {number}
 */
proto.amaas.scan.v1.S2C.prototype.getOffset = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};
/**
 * @param {number} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setOffset = function (value) {
    return jspb.Message.setProto3IntField(this, 3, value);
};
/**
 * optional int32 length = 4;
 * @return {number}
 */
proto.amaas.scan.v1.S2C.prototype.getLength = function () {
    return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};
/**
 * @param {number} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setLength = function (value) {
    return jspb.Message.setProto3IntField(this, 4, value);
};
/**
 * optional string result = 5;
 * @return {string}
 */
proto.amaas.scan.v1.S2C.prototype.getResult = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};
/**
 * @param {string} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setResult = function (value) {
    return jspb.Message.setProto3StringField(this, 5, value);
};
/**
 * repeated int32 bulk_offset = 6;
 * @return {!Array<number>}
 */
proto.amaas.scan.v1.S2C.prototype.getBulkOffsetList = function () {
    return /** @type {!Array<number>} */ (jspb.Message.getRepeatedField(this, 6));
};
/**
 * @param {!Array<number>} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setBulkOffsetList = function (value) {
    return jspb.Message.setField(this, 6, value || []);
};
/**
 * @param {number} value
 * @param {number=} opt_index
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.addBulkOffset = function (value, opt_index) {
    return jspb.Message.addToRepeatedField(this, 6, value, opt_index);
};
/**
 * Clears the list making it empty but non-null.
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.clearBulkOffsetList = function () {
    return this.setBulkOffsetList([]);
};
/**
 * repeated int32 bulk_length = 7;
 * @return {!Array<number>}
 */
proto.amaas.scan.v1.S2C.prototype.getBulkLengthList = function () {
    return /** @type {!Array<number>} */ (jspb.Message.getRepeatedField(this, 7));
};
/**
 * @param {!Array<number>} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setBulkLengthList = function (value) {
    return jspb.Message.setField(this, 7, value || []);
};
/**
 * @param {number} value
 * @param {number=} opt_index
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.addBulkLength = function (value, opt_index) {
    return jspb.Message.addToRepeatedField(this, 7, value, opt_index);
};
/**
 * Clears the list making it empty but non-null.
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.clearBulkLengthList = function () {
    return this.setBulkLengthList([]);
};
/**
 * optional string session_id = 8;
 * @return {string}
 */
proto.amaas.scan.v1.S2C.prototype.getSessionId = function () {
    return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 8, ""));
};
/**
 * @param {string} value
 * @return {!proto.amaas.scan.v1.S2C} returns this
 */
proto.amaas.scan.v1.S2C.prototype.setSessionId = function (value) {
    return jspb.Message.setProto3StringField(this, 8, value);
};
/**
 * @enum {number}
 */
proto.amaas.scan.v1.Stage = {
    STAGE_INIT: 0,
    STAGE_RUN: 1,
    STAGE_FINI: 2
};
/**
 * @enum {number}
 */
proto.amaas.scan.v1.Command = {
    CMD_RETR: 0,
    CMD_QUIT: 1
};
goog.object.extend(exports, proto.amaas.scan.v1);
