// file-security-sdk/lib/protos/scan_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!
'use strict';
Object.defineProperty(exports, "__esModule", { value: true });
var grpc = require('@grpc/grpc-js');
var protos_scan_pb = require('../protos/scan_pb.js');
function serialize_amaas_scan_v1_C2S(arg) {
    if (!(arg instanceof protos_scan_pb.C2S)) {
        throw new Error('Expected argument of type amaas.scan.v1.C2S');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_amaas_scan_v1_C2S(buffer_arg) {
    return protos_scan_pb.C2S.deserializeBinary(new Uint8Array(buffer_arg));
}
function serialize_amaas_scan_v1_S2C(arg) {
    if (!(arg instanceof protos_scan_pb.S2C)) {
        throw new Error('Expected argument of type amaas.scan.v1.S2C');
    }
    return Buffer.from(arg.serializeBinary());
}
function deserialize_amaas_scan_v1_S2C(buffer_arg) {
    return protos_scan_pb.S2C.deserializeBinary(new Uint8Array(buffer_arg));
}
// Interface exported by the server.
var ScanService = exports.ScanService = {
    run: {
        path: '/amaas.scan.v1.Scan/Run',
        requestStream: true,
        responseStream: true,
        requestType: protos_scan_pb.C2S,
        responseType: protos_scan_pb.S2C,
        requestSerialize: serialize_amaas_scan_v1_C2S,
        requestDeserialize: deserialize_amaas_scan_v1_C2S,
        responseSerialize: serialize_amaas_scan_v1_S2C,
        responseDeserialize: deserialize_amaas_scan_v1_S2C,
    },
};
exports.ScanClient = grpc.makeGenericClientConstructor(ScanService, 'Scan');
