// file-security-sdk/lib/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Logger = exports.LogLevel = void 0;
var LogLevel;
(function (LogLevel) {
    LogLevel[LogLevel["OFF"] = 0] = "OFF";
    LogLevel[LogLevel["FATAL"] = 1] = "FATAL";
    LogLevel[LogLevel["ERROR"] = 2] = "ERROR";
    LogLevel[LogLevel["WARN"] = 3] = "WARN";
    LogLevel[LogLevel["INFO"] = 4] = "INFO";
    LogLevel[LogLevel["DEBUG"] = 5] = "DEBUG";
})(LogLevel = exports.LogLevel || (exports.LogLevel = {}));
class Logger {
    constructor() {
        this.logLevel = this.parseLogLevel(process.env.TM_AM_LOG_LEVEL) ?? LogLevel.OFF;
    }
    parseLogLevel(level) {
        switch (level?.toLowerCase()) {
            case 'off':
                return LogLevel.OFF;
            case 'fatal':
                return LogLevel.FATAL;
            case 'error':
                return LogLevel.ERROR;
            case 'warn':
                return LogLevel.WARN;
            case 'info':
                return LogLevel.INFO;
            case 'debug':
                return LogLevel.DEBUG;
            default:
                return undefined;
        }
    }
    shouldLog(level) {
        return level <= this.logLevel;
    }
    log(level, message) {
        if (this.shouldLog(level)) {
            if (this.logCallback !== undefined) {
                this.logCallback(level, message);
            }
            else {
                console.log(`[${LogLevel[level]}] ${message}`);
            }
        }
    }
    debug(message) {
        this.log(LogLevel.DEBUG, message);
    }
    info(message) {
        this.log(LogLevel.INFO, message);
    }
    warn(message) {
        this.log(LogLevel.WARN, message);
    }
    error(message) {
        this.log(LogLevel.ERROR, message);
    }
    fatal(message) {
        this.log(LogLevel.FATAL, message);
    }
    getLogLevel() {
        return this.logLevel;
    }
    setLoggingLevel(level) {
        this.logLevel = level;
    }
    configLoggingCallback(cb) {
        this.logCallback = cb;
    }
}
exports.Logger = Logger;
