// file-security-sdk/lib/scanRun.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScanRun = void 0;
const path_1 = require("path");
const fs_1 = require("fs");
const scanPb = __importStar(require("./protos/scan_pb"));
const utils_1 = require("./utils");
const sha1Prefix = 'sha1:';
const sha256Prefix = 'sha256:';
class ScanRun {
    constructor(scanClient, timeout, logger, tags) {
        this.scanClient = scanClient;
        this.deadline = timeout;
        this.logger = logger;
        this.finalResult = Object.create(null);
        this.tags = tags ?? [];
        this.bulk = false;
    }
    async streamRun(fileName, fileSize, hashes, pml, feedback, verbose, buff) {
        return await new Promise((resolve, reject) => {
            const _deadline = new Date().getTime() + this.deadline * 1000;
            const stream = this.scanClient.run({ deadline: _deadline });
            stream.on('data', (response) => {
                this.handleStreamData(response, fileName, verbose, stream, buff);
            });
            stream.on('error', (err) => {
                const error = ['Metadata key ""'].includes(String(err))
                    ? new Error(`Failed to setup connection to AMaaS host. ${String(err)}`)
                    : err;
                reject(error);
            });
            stream.on('end', () => {
                resolve((verbose) ? this.finalResult : this.finalResult);
            });
            // INIT stage
            const initRequest = new scanPb.C2S();
            initRequest.setStage(scanPb.Stage.STAGE_INIT);
            const fileNameToSet = buff !== undefined ? fileName : (0, path_1.basename)(fileName);
            initRequest.setFileName(fileNameToSet);
            initRequest.setRsSize(fileSize);
            if (this.tags) {
                initRequest.setTagsList(this.tags);
            }
            initRequest.setTrendx(pml);
            initRequest.setSpnFeedback(feedback);
            initRequest.setVerbose(verbose);
            const sha1Digest = hashes[1] ? `${sha1Prefix}${hashes[1]}` : '';
            const sha256Digest = hashes[0] ? `${sha256Prefix}${hashes[0]}` : '';
            this.logger.debug(`sha1: ${sha1Digest}`);
            this.logger.debug(`sha256: ${sha256Digest}`);
            initRequest.setFileSha1(sha1Digest);
            initRequest.setFileSha256(sha256Digest);
            initRequest.setBulk(this.bulk);
            stream.write(initRequest);
        });
    }
    handleStreamData(response, fileName, verbose, stream, buff) {
        const cmd = response.getCmd();
        const stage = response.getStage();
        if (cmd === scanPb.Command.CMD_RETR) {
            let bulkLength = [];
            let bulkOffset = [];
            if (stage !== scanPb.Stage.STAGE_RUN) {
                throw new Error(`Received unexpected command ${cmd} and stage ${stage}.`);
            }
            if (this.bulk) {
                this.logger.debug("enter bulk mode");
                const bulkCount = response.getBulkOffsetList().length;
                if (bulkCount > 1) {
                    this.logger.debug("bulk transfer triggered");
                }
                bulkLength = response.getBulkLengthList();
                bulkOffset = response.getBulkOffsetList();
            }
            else {
                bulkLength = [response.getLength()];
                bulkOffset = [response.getOffset()];
            }
            const fd = buff !== undefined ? undefined : (0, fs_1.openSync)(fileName, 'r');
            for (let i = 0; i < bulkLength.length; i++) {
                this.logger.debug(`stage RUN, try to read ${bulkLength[i]} at offset ${bulkOffset[i]}`);
                const chunk = buff !== undefined ? buff.subarray(bulkOffset[i], bulkOffset[i] + bulkLength[i]) : Buffer.alloc(bulkLength[i]);
                if (fd !== undefined) {
                    (0, fs_1.readSync)(fd, chunk, 0, bulkLength[i], bulkOffset[i]);
                }
                const request = new scanPb.C2S();
                request.setStage(scanPb.Stage.STAGE_RUN);
                request.setOffset(bulkOffset[i]);
                request.setChunk(chunk);
                stream.write(request);
            }
            if (fd !== undefined) {
                (0, fs_1.closeSync)(fd);
            }
        }
        else if (cmd === scanPb.Command.CMD_QUIT) {
            this.logger.debug('receive QUIT, exit loop...\n');
            const result = response.getResult();
            const resultJson = JSON.parse(result);
            this.finalResult = resultJson;
            stream.end();
        }
        else {
            this.logger.debug('unknown command...');
            stream.end();
        }
    }
    async scanFile(name, size, pml, feedback, verbose, digest) {
        const hashes = digest ? await (0, utils_1.getHashes)(name, ['sha256', 'sha1'], 'hex') : ['', ''];
        return await this.streamRun(name, size, hashes, pml, feedback, verbose)
            .then(result => {
            return result;
        })
            .catch(reason => { throw reason; });
    }
    async scanBuffer(name, buff, pml, feedback, verbose, digest) {
        const size = Buffer.byteLength(buff);
        const hashes = digest ? await (0, utils_1.getBufferHashes)(buff, ['sha256', 'sha1'], 'hex') : ['', ''];
        return await this.streamRun(name, size, hashes, pml, feedback, verbose, buff)
            .then(result => result)
            .catch(err => { throw err; });
    }
}
exports.ScanRun = ScanRun;
